### PR TITLE
tests: subsys: bootloader: add bootloader tag

### PR DESCRIPTION
**Pull request by nordic-piks**
Extend tags and depencies.

### CODE REVIEW

**Comment by nordic-piks** [comment]
The idea behind this change is to allow automatic detection of PRs not related to given code area.
In this case, there is defined tag and files/folder with source code related to that area.
This list of files/folders is then compared to files changed in PR. If particular PR does not change them,
tests related to specified tag should not be executed - will be excluded.

To gather this list, I analyzed all source files which are compiled for this samples/tests.

**Review by nordic-segl — APPROVED** [review]

File: scripts/ci/tags.yaml [hunk 1]
```diff
@@ -516,3 +516,23 @@ peripheral_802154:
...
+    - nrf/lib/fprotect/
+    - nrf/lib/hw_unique_key/
+    - nrf/lib/identity_key/
+    - nrf/lib/modem_antenna/
+    - nrf/lib/nrf_modem_lib/
+    - nrf/samples/bootloader/
+    - nrf/subsys/bootloader/
+    - nrf/subsys/fw_info/
+    - nrf/subsys/nrf_security/
+    - nrf/tests/subsys/bootloader/
```
Comment (gchwier) [inline scripts/ci/tags.yaml:535]:
```suggestion
    - nrf/tests/modules/mcuboot/
    - nrf/tests/subsys/bootloader/
    - nrf/tests/subsys/dfu/
    - nrf/tests/drivers/fprotect/
    - nrf/tests/drivers/flash/
    - nrf/samples/nrf5340/extxip_smp_svr/
```

File: scripts/ci/tags.yaml [hunk 2]
```diff
@@ -516,3 +516,23 @@ peripheral_802154:
...
+    - nrfxlib/crypto/
+    - nrfxlib/nrf_modem/
+    - zephyr/soc/nordic/
```
Comment (gchwier) [inline scripts/ci/tags.yaml:538]:
```suggestion
    - zephyr/soc/nordic/
    - zephyr/samples/sysbuild/with_mcuboot/
    - zephyr/tests/subsys/dfu/
    - zephyr/tests/boot/test_mcuboot/
```

Comment (gchwier) [reply scripts/ci/tags.yaml:538]: what you think about adding mcumgr also here?
```
    - zephyr/subsys/mgmt/mcumgr/
    - zephyr/tests/subsys/mgmt/mcumgr/
    - zephyr/samples/subsys/mgmt/mcumgr/
    - zephyr/include/zephyr/mgmt/mcumgr/
```
to run all bootloader CI if any change in mcumgr will be done?

File: scripts/ci/tags.yaml (hunk 1 above)
Comment (nordic-piks) [reply scripts/ci/tags.yaml:535]: Not really.
Those depencies are related to tests/subsys/bootloader only.
So this is a list for this specific tests not for all bootloader tests and samples.
Thus adding other tests does not fit the idea.

File: scripts/ci/tags.yaml (hunk 2 above)
Comment (nordic-piks) [reply scripts/ci/tags.yaml:538]: The same as above, those paths are depencies for tests/subsys/bootloader.
Maybe I should make this tag name more specific?
Intention is not to mark all bootloader/dfu etc. tests.

File: scripts/ci/tags.yaml (hunk 1 above)
Comment (nordic-piks) [reply scripts/ci/tags.yaml:535]: For suggested, I will propose to add separated tag e.g dfu, fprotect flash etc.
But this is different PR.

File: scripts/ci/tags.yaml (hunk 2 above)
Comment (nordic-piks) [reply scripts/ci/tags.yaml:538]: For those proposed I will suggest to create different tag mgmt or dfu but this is outside of this PR.

**Review by gchwier — APPROVED** [review]

**Pull request merged**